### PR TITLE
RIE and BEDROC implementation

### PR DESCRIPTION
**Pull request by marta-sd**
We get the same results as RDKit, so I assume we calculate it correctly

### CODE REVIEW

File: oddt/metrics.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (mwojcikowski) [inline oddt/metrics.py]: Why do we need to declare the encoding? I prefer not to define it, rather remove the UTF8 characters ( I can see hyphens in citations which might easily be replaced by ASCII chars).

File: oddt/metrics.py [hunk 2]
```diff
@@ -190,3 +191,92 @@ def standard_deviation_error(y_true, y_pred):
...
+    Returns
+    -------
+    rie : float
+         Robust Initial Enhancement
+
+    Notes
+    ----------
+    .. [1] Sheridan, R. P.; Singh, S. B.; Fluder, E. M.; Kearsley, S. K.
+           Protocols for bridging the peptide to nonpeptide gap in topological
+           similarity searches. J. Chem. Inf. Comput. Sci. 2001, 41, 1395−1406.
```
Comment (mwojcikowski) [inline oddt/metrics.py]: Non-ASCII hypen

File: oddt/metrics.py [hunk 3]
```diff
@@ -190,3 +191,92 @@ def standard_deviation_error(y_true, y_pred):
...
+    if pos_label is None:
+        pos_label = 1
+    labels = y_true == pos_label
+    alpha = float(alpha)
+    N = float(len(labels))
+    n = float(labels.sum())
+    r = np.argwhere(labels).astype(float) + 1  # need 1-based ranking
+    x = r / N
+    ra = float(n) / float(N)
+    return (np.exp(-alpha * x).sum()
```
Comment (mwojcikowski) [inline oddt/metrics.py]: Can we keep the return statements short? We can create an out variable and return it.

File: oddt/metrics.py [hunk 4]
```diff
@@ -190,3 +191,92 @@ def standard_deviation_error(y_true, y_pred):
...
+    Returns
+    -------
+    bedroc : float
+        Boltzmann-Enhanced Discrimination of Receiver Operating Characteristic
+
+    Notes
+    ----------
+    .. [1] Truchon J-F, Bayly CI. Evaluating virtual screening methods: good
+           and bad metrics for the “early recognition” problem.
+           J Chem Inf Model. 2007;47: 488–508.
```
Comment (mwojcikowski) [inline oddt/metrics.py]: Non-ASCII hyphen and possibly quotes.

File: oddt/metrics.py [hunk 5]
```diff
@@ -190,3 +191,92 @@ def standard_deviation_error(y_true, y_pred):
...
+    if pos_label is None:
+        pos_label = 1
+    labels = y_true == pos_label
+    alpha = float(alpha)
+    N = float(len(labels))
+    n = float(labels.sum())
+    ra = float(n) / float(N)
+    ri = 1 - ra
+    rie_score = rie(y_true, y_score, alpha=alpha, pos_label=pos_label)
+    return (rie_score * ra * np.sinh(alpha / 2) / (np.cosh(alpha / 2)
```
Comment (mwojcikowski) [inline oddt/metrics.py]: Complex return statement, which could be simplified

File: oddt/metrics.py (hunk 1 above)
Comment (marta-sd) [reply oddt/metrics.py]: I don't know why we need encoding, you've written this line :) I think it was added automatically at some point, lets just remove it and fix all non-ascii characters

**Pull request merged**